### PR TITLE
Add streamline Logger

### PR DESCRIPTION
**Pull request by DavidLegg**
* **Tickets addressed:** N/A
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
Adds a Logger class to the streamline framework, which uses topics to emit log messages directly as events. Also builds such a Logger when the Registrar is initialized, and reconfigures errors to go to that logger when the error behavior is set to LOG. Before this, errors were collected awkwardly into a discrete string resource.

## Verification

Since the logger is connected to the Registrar's error behavior, this was tested using the streamline-demo example model. By adding "CauseError" activities to the plan, we can trip the resource error handling machinery, and verify that appropriate error messages appear in the Simulation Events tab.

image

## Documentation

None yet - #1494 should probably be updated to tell users how to use the logger in their model/activities.

This does remove the `errors` resource, but it leaves the `numberOfErrors` resource intact, which we should note for any users who may have depended on `errors`.

## Future work

None planned

### CODE REVIEW

File: contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/debugging/Logger.java
Comment (mattdailis) [inline contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/debugging/Logger.java]: @DavidLegg and I discussed that this TODO comment should be addressed before merging this PR. In particular - there's no need to explicitly include the log timestamp in the message, because every serializable event is time-tagged

**Review by mattdailis — CHANGES_REQUESTED** [review]
See https://github.com/NASA-AMMOS/aerie/pull/1500/files#r1705853363

File: contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/modeling/Registrar.java
Comment (mattdailis) [inline contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/modeling/Registrar.java]: If we can avoid changing the signature of the Registrar constructor, I think this PR can be backwards compatible with code that already uses the streamline framework

File: contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/modeling/Registrar.java
Comment (mattdailis) [inline contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/modeling/Registrar.java:63]: In the event that users have written constraints that depend on the previous resource definition, it may be worth calling out this change in the release notes. Not a reason not to make this change, but worth communicating explicitly.

It also highlights that constraints (and scheduling?) would benefit from being able to refer to simulation events.

**Review by mattdailis — APPROVED** [review]
Thanks for making this backwards compatible! Looks good to me

EDIT: Resolving conversation below so it doesn't block merge, but copying up here for visibility:
> In the event that users have written constraints that depend on the previous resource definition, it may be worth calling out in the release notes [that the errors resources has been replaced with a numberOfErrors resource]. Not a reason not to make this change, but worth communicating explicitly.

**Comment by DavidLegg** [comment]
One slight correction to that note on the `errors` and `numberOfErrors` resources - We actually had `numberOfErrors` already, just derived in a different way. This PR does remove `errors` itself, though, so I agree we should put that in the release notes just in case.

**Pull request merged**
